src/components/BreadcrumbView/BreadcrumbView.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from 'react';
+import { Paths } from '../../config';
+import { Link } from 'react-router-dom';
+import { Breadcrumb } from 'patternfly-react';
+import { ActiveFilter } from '../../types/Filters';
+import { FilterSelected } from '../Filters/StatefulFilters';
+import { dicIstioType } from '../../types/IstioConfigList';
+
+interface BreadCumbViewProps {
+  location: {
+    pathname: string;
+    search: string;
+  };
+}
+
+interface BreadCumbViewState {
+  namespace: string;
+  itemName: string;
+  item: string;
+  pathItem: string;
+  tab: string;
+  istioType?: string;
+}
+
+const ItemNames = {
+  applications: 'App',
+  services: 'Service',
+  workloads: 'Workload',
+  istio: 'Istio Object'
+};
+
+const IstioName = 'Istio Config';
+
+export class BreadcrumbView extends React.Component<BreadCumbViewProps, BreadCumbViewState> {
+  static capitalize = (str: string) => {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+  };
+
+  static getTab = (location: string) => {
+    const urlParams = new URLSearchParams(location);
+    switch (urlParams.get('tab')) {
+      case 'info':
+        return 'Info';
+        break;
+      case 'metrics':
+        return 'Metrics';
+        break;
+      case 'in_metrics':
+        return 'Inbound Metrics';
+        break;
+      case 'out_metrics':
+        return 'Outbound Metrics';
+        break;
+      default:
+        return 'Info';
+    }
+  };
+
+  constructor(props: BreadCumbViewProps) {
+    super(props);
+    this.state = this.updateItem();
+  }
+
+  updateItem = () => {
+    const namespaceRegex = /namespaces\/([a-z0-9\-]+)\/([a-z0-9\-]+)\/([a-z0-9\-]+)(\/([a-z0-9\-]+))?/;
+    const match = this.props.location.pathname.match(namespaceRegex) || [];
+    const ns = match[1];
+    const page = Paths[match[2].toUpperCase()];
+    const istioType = match[3];
+    let itemName = match[3];
+    if (page === 'istio') {
+      itemName = match[4];
```

Comment:
Take a double look here: this could not be true as istio has two types of urls:
http://localhost:3000/console/namespaces/istio-system/istio/templates/logentries/accesslog
http://localhost:3000/console/namespaces/istio-system/istio/policies/disable-mtls-jaeger-collector

Templates and Adapters have subtype but not others objects.

Reply:
let me check it